lib/matplotlib/widgets.py

```diff
@@ -1010,20 +1013,27 @@ def __init__(self, ax, label, initial='',
             The color of the box when the mouse is over it.
         label_pad : float
             The distance between the label and the right side of the textbox.
+        textalignment : {'left', 'center', 'right'}
+            The horizontal location of the text.
         """
         super().__init__(ax)
 
-        self.DIST_FROM_LEFT = .05
+        self._DIST_FROM_LEFT = .05
+
+        self._text_position = _api.check_getitem(
+                                {"left": 0.05,
+                                 "center": 0.5,
+                                 "right": 0.95}, key=textalignment)
```

Comment:
The keyword name must match to produce the right exception.
```suggestion
        self._text_position = _api.check_getitem(
            {"left": 0.05, "center": 0.5, "right": 0.95},
            textalignment=textalignment)
```